### PR TITLE
Jonee 17 feat: Add SavingsAccount and MoneyOrder enhancements to ATM simulator

### PR DESCRIPTION
**Pull request by jmckellar252-png**
Summary:
 Added SavingsAccount and MoneyOrder enhancements to ATM simulator without adding new public methods.

Changes Made:

SavingsAccount: 
New account type that works like CheckingAccount but designed for savings
MoneyOrder: 
New monetary tool that withdraws funds immediately upon creation (unlike Check which withdraws on deposit)
Polymorphism:
Created Account interface so BankAtm can handle multiple account types
BankAtm Integration: 
Added overloaded depositFunds() methods for new instruments

Technical Details:

- Used method overloading instead of adding new public methods
- All existing functionality preserved (27 original tests still pass)
- Added comprehensive tests (now 48+ total tests passing)
- Maintains 80%+ code coverage requirement

Assignment Requirements Met:
- At least two functional enhancements implemented
- No new public methods added to BankAtm
- Updated tests and documentation
- Build successful with all tests passing

Files Added:
 Account.java, SavingsAccount.java, MoneyOrder.java + corresponding test files
Files Modified: 
BankAtm.java, CheckingAccount.java, Customer.java, Check.java, BankAtmTest.java

### CODE REVIEW

**Review by anthonydmays — COMMENTED** [review]
## 🎓 Automated Grading Report

**Student:** Jonee Mckellar
**Date:** 9/19/2025

### Feedback
The submission demonstrates a good understanding of the SOLID principles by integrating the `SavingsAccount` and `MoneyOrder` functionalities into the existing `BankAtm` class without adding new public methods. The use of an `Account` interface to generalize account handling is a solid design choice, allowing for the addition of different account types. However, the implementation does not fully address the requirement for an `AuditLog` to track all transactions, which is a key component of the functional requirements. 

Technically, the code is well-structured and adheres to Java coding conventions. The changes made to existing methods are consistent with the existing style, and the use of interfaces promotes code reusability and adherence to the Open/Closed principle. The unit tests provided cover the new functionalities, but they could be more comprehensive in testing edge cases and integration with the `BankAtm` class. 

Overall, the submission meets most of the functional and technical requirements, but could be improved by implementing the transaction logging feature and enhancing test coverage.

---
*This is an automated preliminary review. Please review and adjust before finalizing.*

File: lesson_17/bank/bank_app/src/main/java/com/codedifferently/lesson17/bank/SavingsAccount.java [hunk 1]
```diff
@@ -0,0 +1,105 @@
+package com.codedifferently.lesson17.bank;
+
+import com.codedifferently.lesson17.bank.exceptions.InsufficientFundsException;
+import java.util.Set;
+
+/** Represents a savings account that doesn't allow check operations. */
+public class SavingsAccount implements Account {
```
Comment (anthonydmays) [inline lesson_17/bank/bank_app/src/main/java/com/codedifferently/lesson17/bank/SavingsAccount.java:7]: Instead of re-implementing account logic across `SavingsAccount` and `CheckingAccount`, you could implement a base class that allow both to share the logic.

**Review by anthonydmays — APPROVED** [review]
